contrib_bots/lib/gmail.py

```diff
@@ -0,0 +1,156 @@
+#instructions to install googleapiclient can be found in this bot's readme file.
+from __future__ import print_function
+import base64
+import httplib2
+import logging
+import mimetypes
+import os
+import re
+
+from googleapiclient import discovery, errors
+from email.mime.text import MIMEText
+from oauth2client.file import Storage
+
+class GmailHandler(object):
+    '''
+    This plugin facilitates sending emails from a Gmail account
+    from within Zulip. It looks for messages starting with '@gmail'.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to send emails from a
+            personal gmail account. Users should preface
+            messages with "@gmail".
+
+            Enter messages in the following format:
+            "@gmail <recipient_email> <subject> <body> --label <label>"
+            The --label tag is optional and can be left out.
+
+            Before running this, make sure to create a bot and store its information
```

Comment:
You can omit these two lines, because if somebody sees this message, they will already have supplied a zuliprc file.